File: Ejercicio 9/ejercicio9.java
import java.util.ArrayList;
import java.util.List;

class Facultad {
    private String nombre;

    public Facultad(String nombre) {
        this.nombre = nombre;
    }

    public String getNombre() { return nombre; }
}

class Universidad {
    private String nombreUni;
    private List<Facultad> facultades;

    public Universidad(String nombre) {
        this.nombreUni = nombre;
        this.facultades = new ArrayList<>();
    }

  
    public void agregarFacultad(Facultad f) {
        this.facultades.add(f);
    }

    public void mostrarEstructura() {
        System.out.println("Universidad: " + nombreUni);
        for (Facultad f : facultades) {
            System.out.println(" -> Facultad de " + f.getNombre());
        }
    }
}

public class Main {
    public static void main(String[] args) {
      
        Facultad f1 = new Facultad("Ingeniería");
        Facultad f2 = new Facultad("Ciencias de la Salud");

        Universidad miUni = new Universidad("Universidad Nacional");

     
        miUni.agregarFacultad(f1);
        miUni.agregarFacultad(f2);

      
        miUni.mostrarEstructura();
    }
}

File: Ejercicio 9/ejercicio9.py
class Facultad:
    def __init__(self, nombre):
        self.nombre = nombre

class Universidad:
    def __init__(self, nombre_uni):
        self.nombre_uni = nombre_uni
        self.facultades = []

   
    def incluir_facultad(self, facultad):
        self.facultades.append(facultad)


if __name__ == "__main__":
    
    fac_ia = Facultad("Inteligencia Artificial")
    fac_derecho = Facultad("Derecho")

    
    u_distrital = Universidad("U. Distrital")

  
    u_distrital.incluir_facultad(fac_ia)
    u_distrital.incluir_facultad(fac_derecho)

 
    print(f"Estructura de la {u_distrital.nombre_uni}:")
    for f in u_distrital.facultades:
        print(f"  - {f.nombre}")
